🎉 New Source: Northpass LMS

sajarin opened the pull request:
## What
Adding new connector for [Northpass LMS](https://developers.northpass.com/docs/) 🎉 


## How
Uses low code CDK to implement three streams: people, courses and course enrollments

## Review guide
1. manifest.yaml
2. metadata.yaml
3. docs/integrations/sources/northpass-lms.md

## User Impact
New connector available for all users

## Can this PR be safely reverted and rolled back?
<!--
* If unsure, leave it blank.
-->
- [X] YES 💚

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/metadata.yaml:
```diff
@@ -0,0 +1,35 @@
+data:
+  allowedHosts:
+    hosts:
+      - api.northpass.com/v2
```
```suggestion
      - api.northpass.com
```

Just the hostname is fine, let's keep it simple.

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/metadata.yaml:
```diff
@@ -0,0 +1,35 @@
+  dockerImageTag: 0.1.0
+  dockerRepository: airbyte/source-northpass-lms
+  githubIssueLabel: source-northpass-lms
+  icon: northpass-lms.svg
```
```suggestion
  icon: icon.svg
```

natikgadzhi replied:
Rename the icon file, too. I realize this is a bizarre request, but I promise it'll make sense in a week ;)

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/pyproject.toml:
```diff
@@ -0,0 +1,29 @@
+
+[tool.poetry.dependencies]
+python = "^3.9,<3.12"
+airbyte-cdk = "^0"
```
```suggestion
airbyte-cdk = "^4.3.0"
```

natikgadzhi replied:
We've published _a few_ new versions of the CDK —let's upgrade to the latest one!

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/source_northpass_lms/manifest.yaml:
```diff
@@ -0,0 +1,768 @@
+version: "0.29.0"
```
```suggestion
version: "4.3.0"
```

Wow, this was old. Started with a generator? I need to clean this up today huh.

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/source_northpass_lms/manifest.yaml:
```diff
@@ -0,0 +1,768 @@
+      backoff_strategies:
+        - type: ConstantBackoffStrategy
+          backoff_time_in_seconds: 5
+      response_filters:
```
Glad to see error handlers in a new connector! 👏🏼

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/integration_tests/acceptance.py:
```diff
@@ -0,0 +1,16 @@
+#
```
@sajarin delete this file please. If it's just boilerplate, no need to keep it.

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/README.md:
```diff
@@ -0,0 +1,103 @@
+
+### Prerequisites
+
+* Python (`^3.9`)
```
```suggestion
* Python (`^3.10`)
```

natikgadzhi reviewed (approved):
This looks good. Putting an approval here, even if merged as is that's ok.

Left comments though:
- Remove boilerplate file in acceptance_tests.py
- Move icon to be just icon.svg
- Bump to CDK 4.3.0 and run `poetry lock`

Accept suggestions, after that, let's format and @marcosmarxm can merge this.

sajarin commented:
Thanks for the review @natikgadzhi, addressed the comments above. Could I convince you to run the format command on your instance, for some reason it takes a long time on my local? 

Appreciate the comments, in return, happy to provide some feedback afterwards on the whole process from start to finish. I ended up developing two connectors (this one and the hibob one) in two different ways (starting from the generator vs starting from the low-code builder)

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/README.md:
````diff
@@ -0,0 +1,103 @@
+poetry run source-northpass-lms discover --config secrets/config.json
+poetry run source-northpass-lms read --config secrets/config.json --catalog sample_files/configured_catalog.json
+```
+
+### Running tests
+
+To run tests locally, from the connector directory run:
+
+```
+poetry run pytest tests
+```
````
```suggestion
```

Since there are no unit tests in the low-code connector, this block does not apply.

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/integration_tests/__init__.py:
```diff
@@ -0,0 +1,3 @@
+#
```
Since there are no python files in the directory, this file is redundant.

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/main.py:
```diff
@@ -0,0 +1,8 @@
+from source_northpass_lms.run import run
+
+if __name__ == "__main__":
+    run()
```
nice!

natikgadzhi commented:
Automerge on, running the CI.

natikgadzhi commented:
> Appreciate the comments, in return, happy to provide some feedback afterwards on the whole process from start to finish. I ended up developing two connectors (this one and the hibob one) in two different ways (starting from the generator vs starting from the low-code builder)

I'm very interested. Please post — either here, but let's also talk more on Slack. 

Would you be ineterested in contributing connectors _straight from Builder_ if it had a button to make a PR? Because it has, it's under a feature flag and I'm writing docs for it today.

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/metadata.yaml:
```diff
@@ -0,0 +1,35 @@
+  icon: icon.svg
+  license: MIT
+  name: Northpass LMS
+  releaseDate: 08-06-2024
```
```suggestion
  releaseDate: 2024-08-06
```

natikgadzhi commented on airbyte-integrations/connectors/source-northpass-lms/metadata.yaml:
```diff
@@ -0,0 +1,35 @@
+    # Please update to the latest version of the connector base image.
+    # https://hub.docker.com/r/airbyte/python-connector-base
+    # Please use the full address with sha256 hash to guarantee build reproducibility.
+    baseImage: docker.io/airbyte/python-connector-base:2.0.0@sha256:c1b339a44caa3e060fb0df4bea89a11d642c86f67618aa0c72a03855772466d9
```
```suggestion
    baseImage: docker.io/airbyte/python-connector-base:2.0.0@sha256:c44839ba84406116e8ba68722a0f30e8f6e7056c726f447681bb9e9ece8bd916
```
